src/server/handlers/consents/{ID}/revoke.ts

```diff
@@ -0,0 +1,99 @@
+/* istanbul ignore file */
+
+/*
+ * This flag is to ignore BDD testing for model
+ * which will be addressed in the future in
+ * ticket #354
+ */
+
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop
+ files are distributed onan 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+ --------------
+ ******/
+import {
+  patchConsentRevoke,
+  isConsentRequestInitiatedByValidSource,
+  revokeConsentStatus
+} from '../../../../domain/consents/revoke'
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import Logger from '@mojaloop/central-services-logger'
+import { Enum } from '@mojaloop/central-services-shared'
+import { consentDB } from '../../../../lib/db'
+import { Consent } from '../../../../model/consent'
+
+export async function validateRequestAndRevokeConsent (
+  request: Request): Promise<void> {
+  const consentId = request.params.id
+
+  try {
+    // Fetch consent from database using ID
+    let consent: Consent
+    try {
+      consent = await consentDB.retrieve(consentId)
+    } catch (error) {
+      Logger.push(error)
+      Logger.error('Error in retrieving consent')
+
+      // If consent cannot be retrieved using given ID, send PUT ...error back
+      // TODO: Error Handling dealt with in future ticket #355
+      throw (new Error('NotImplementedYetError'))
+    }
+
+    // If request is not intiated by valid source, send PUT ...error back
+    if (!isConsentRequestInitiatedByValidSource(consent, request)) {
```

Comment:
I think this is handled by the OpenAPI-Hapi plugin.

Reply:
nope